File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/barrierfree/BarrierfreeContactAdapter.java
package de.tum.in.tumcampusapp.component.ui.barrierfree;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.text.util.Linkify;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import de.tum.in.tumcampusapp.R;
import de.tum.in.tumcampusapp.component.other.generic.adapter.SimpleStickyListHeadersAdapter;
import de.tum.in.tumcampusapp.component.tumui.person.PersonsDetailsActivity;
import de.tum.in.tumcampusapp.component.tumui.person.model.Person;
import de.tum.in.tumcampusapp.component.ui.barrierfree.model.BarrierfreeContact;
import se.emilsjolander.stickylistheaders.StickyListHeadersAdapter;

/**
 * An adapter used to display contact information in barrierfree page.
 */
public class BarrierfreeContactAdapter extends SimpleStickyListHeadersAdapter<BarrierfreeContact>
        implements StickyListHeadersAdapter {
    public BarrierfreeContactAdapter(Context context, List<BarrierfreeContact> infos) {
        super(context, infos);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        final ViewHolder holder;
        View view = convertView;

        if (view == null) {
            // Crate UI element
            view = getInflater().inflate(R.layout.activity_barrier_free_contact_listview, parent, false);
            holder = new ViewHolder(view, parent);
            view.setTag(holder);
        } else {
            holder = (ViewHolder) view.getTag();
        }

        // display information of current person
        final BarrierfreeContact contact = getInfoList().get(position);

        if (!contact.isValid()) {
            view.setVisibility(View.GONE);
            return view;
        }

        // set Information
        view.setVisibility(View.VISIBLE);
        holder.setContent(contact);

        return view;
    }

    // the layout of the list
    class ViewHolder {
        TextView name;
        TextView phone;
        TextView email;
        TextView more;

        ViewHolder(View view, ViewGroup parent) {

            name = view.findViewById(R.id.barrierfreeContactListViewName);
            phone = view.findViewById(R.id.barrierfreeContactListViewPhone);
            email = view.findViewById(R.id.barrierfreeContactListViewEmail);
            more = view.findViewById(R.id.barrierfreeContactListViewTumOnlineMore);
        }

        void setContent(final BarrierfreeContact contact) {
            name.setText(contact.getName());
            phone.setText(contact.getTelephone(), TextView.BufferType.SPANNABLE);
            Linkify.addLinks(phone, Linkify.ALL);
            email.setText(contact.getEmail());

            // Has information in tumonline
            if (!contact.isHavingTumID()) {
                more.setVisibility(View.GONE);
            } else {
                // Jump to PersonDetail Activity
                more.setVisibility(View.VISIBLE);
                more.setText(context.getString(R.string.more_info));
                more.setOnClickListener(v -> {
                    Person person = new Person();
                    person.setName(contact.getName());
                    person.setId(contact.getTumID());

                    Bundle bundle = new Bundle();
                    bundle.putSerializable("personObject", person);

                    Intent intent = new Intent(context, PersonsDetailsActivity.class);
                    intent.putExtras(bundle);
                    context.startActivity(intent);
                });
            }
        }
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/barrierfree/BarrierFreeContactAdapter.kt
package de.tum.`in`.tumcampusapp.component.ui.barrierfree

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.text.util.Linkify
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import de.tum.`in`.tumcampusapp.R
import de.tum.`in`.tumcampusapp.component.other.generic.adapter.SimpleStickyListHeadersAdapter
import de.tum.`in`.tumcampusapp.component.tumui.person.PersonDetailsActivity
import de.tum.`in`.tumcampusapp.component.tumui.person.model.Person
import de.tum.`in`.tumcampusapp.component.ui.barrierfree.model.BarrierFreeContact
import se.emilsjolander.stickylistheaders.StickyListHeadersAdapter

/**
 * An adapter used to display contact information in barrierfree page.
 */
class BarrierFreeContactAdapter(context: Context, infos: List<BarrierFreeContact>) :
        SimpleStickyListHeadersAdapter<BarrierFreeContact>(context, infos.toMutableList()), StickyListHeadersAdapter {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val holder: ViewHolder
        val view: View
        if (convertView == null) {
            // Crate UI element
            view = inflater.inflate(R.layout.activity_barrier_free_contact_listview, parent, false)
            holder = ViewHolder(view)
            view.tag = holder
        } else {
            view = convertView
            holder = view.tag as ViewHolder
        }

        // display information of current person
        val contact = itemList[position]

        if (!contact.isValid) {
            view.visibility = View.GONE
            return view
        }

        // set Information
        view.visibility = View.VISIBLE
        holder.setContent(contact)

        return view
    }

    // the layout of the list
    internal inner class ViewHolder(view: View) {
        var name: TextView = view.findViewById(R.id.barrierfreeContactNameTextView)
        var phone: TextView = view.findViewById(R.id.barrierfreeContactPhoneTextView)
        var email: TextView = view.findViewById(R.id.barrierfreeContactEmailTextView)
        var more: TextView = view.findViewById(R.id.barrierfreeContactMoreTextView)

        fun setContent(contact: BarrierFreeContact) {
            name.text = contact.name
            phone.setText(contact.telephone, TextView.BufferType.SPANNABLE)
            Linkify.addLinks(phone, Linkify.ALL)
            email.text = contact.email

            if (!contact.hasTumID) {
                more.visibility = View.GONE
            } else {
                // Jump to PersonDetail Activity
                more.visibility = View.VISIBLE
                more.text = context.getString(R.string.more_info)
                more.setOnClickListener {
                    val person = Person(name = contact.name, id = contact.tumID)
                    val bundle = Bundle().apply {
                        putSerializable("personObject", person)
                    }
                    val intent = Intent(context, PersonDetailsActivity::class.java).apply {
                        putExtras(bundle)
                    }
                    context.startActivity(intent)
                }
            }
        }
    }
}